src/library/process.js

```diff
@@ -165,6 +230,33 @@ export const processMessage = async (AWS, region, service, account,
     }
   }
 
+  if (itemExists(procRes.workerResp, 'sharedMicroApplicationUserData')) {
+    if (typeof procRes.workerResp.sharedMicroApplicationUserData !== 'object') {
```

Comment:
Add validation that `procRes.workerResp.sharedMicroApplicationUserData` has correct shape beyond object (microApplicationsToShareWith and serviceData keys)?

I think for serviceUserData & serviceAccountData in some cases workers return what ever their inputs were as a kind of "just-in-case" but that would cause problems here since I think the worker's input would have top-level keys that are service names

Reply:
Good callout, will do this.